src/Nethermind/Nethermind.Trie/Pruning/ScopedTrieStore.cs

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Trie.Pruning;
+
+public class ScopedTrieStore : IScopedTrieStore
+{
+    private ITrieStore _trieStoreImplementation;
+    private readonly Hash256? _address;
+
+    public ScopedTrieStore(ITrieStore fullTrieStore, Hash256? address)
+    {
+        _trieStoreImplementation = fullTrieStore;
+        _address = address;
+    }
+
+    public TrieNode FindCachedOrUnknown(in TreePath path, Hash256 hash)
+    {
+        return _trieStoreImplementation.FindCachedOrUnknown(_address, path, hash);
+    }
+
+    public byte[]? LoadRlp(in TreePath path, Hash256 hash, ReadFlags flags = ReadFlags.None)
+    {
+        return _trieStoreImplementation.LoadRlp(_address, path, hash, flags);
+    }
+
+    public byte[]? TryLoadRlp(in TreePath path, Hash256 hash, ReadFlags flags = ReadFlags.None)
```

Comment:
hash by ref using `in Hash256` to copy less?

Reply:
Does it make a difference if its a reference type?

Reply:
Oh, I missed it! Solved!